app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetController.kt

```diff
@@ -488,6 +490,7 @@ class BloodPressureEntrySheetController @Inject constructor(
           bloodPressureRepository
               .saveMeasurement(newBp.patientUuid, newBp.systolic, newBp.diastolic, toUtcInstant(newBp.date))
               .map { BloodPressureSaved(wasDateChanged = prefilledDate != newBp.date) }
+              .doOnSuccess { appointmentRepository.markAppointmentsCreatedBeforeTodayAsVisited(newBp.patientUuid) }
```

Comment:
Since `AppointmentRepository#markAppointmentsCreatedBeforeTodayAsVisited` returns a `Completable`, invoking this in a `.doOnSuccess` is not the right solution since the `Completable` is never subscribed to. In addition, this is not a side effect, but a flow that must be executed as a part of this chain. What you need is a `.flatMap` instead:

```kotlin
.flatMap { bpSaved -> 
  appointmentRepository
    .markAppointmentsCreatedBeforeTodayAsVisited(newBp.patientUuid)
    .toSingleDefault(bpSaved)
}
```

Reply:
I guess I am still not sure what side effects are for. The reason I put this as a side effect was because it is not the main job in this chain. I will change this then.

Reply:
I've found that an easy way is to ask yourself, "if this fails, does it impact the behaviour of the app in a way that the user should be informed?". If yes, it should not be a side effect.

Reply:
Oh! Let's talk about this in person.